lib/ui/screens/profile/task_details.dart

```diff
@@ -212,45 +210,76 @@ Widget _buildBody(context, Task task, Grihini grihini) {
   );
 }
 
-Widget taskCard() {
-  return Padding(
-    padding: EdgeInsets.all(w * 0.01),
-    child: Container(
-      height: h / 8,
-      width: w / 3.5,
-      decoration: BoxDecoration(
-        borderRadius: BorderRadius.all(Radius.circular(w * 0.02)),
-        gradient: LinearGradient(
-          colors: [
-            AppColors.lightYellow,
-            AppColors.darkYellow,
-          ],
+taskListGenerator(taskList,context) {
+  List<Widget> taskCards = [];
+  for (Task task in taskList) {
+    taskCards.add(taskCard(task,context));
+  }
+  return Row(
+    children: taskCards,
+  );
+}
+
+Widget taskCard(Task task,BuildContext context) {
+  return GestureDetector(
+    onTap: (){
+      currentTask = task;
+      String oderStatusRoute = taskAcceptScreenRoute;
+      switch (task.orderStatus) {
+        case OrderStatus.CREATED:
+          oderStatusRoute = taskAcceptScreenRoute;
+          break;
+        case OrderStatus.GROCERY_DROP_OFF:
+          oderStatusRoute = groceryPendingScreenRoute;
+          break;
+        case OrderStatus.READY_FOR_PICKUP:
+          oderStatusRoute = achaarPreparedScreenRoute;
+          break;
+        case OrderStatus.ORDER_COMPLETED:
+          oderStatusRoute = achaarPreparedScreenRoute;
```

Comment:
Why are READY_FOR_PICKUP and ORDER_COMPLETED the same screen route.
How do we differentiate the UI we show for these two?

Reply:
I havent completed the ui for order completed state. I''ll change it by preparing a dummy page.